collector/generatorreceiver/internal/topology/kubernetes.go

```diff
@@ -0,0 +1,160 @@
+package topology
+
+import (
+	"math"
+	"math/rand"
+	"time"
+)
+
+type Kubernetes struct {
+	ClusterName string   `json:"cluster_name" yaml:"cluster_name"`
+	Request     Resource `json:"request" yaml:"request"`
+	Limit       Resource `json:"limit" yaml:"limit"`
+
+	ReplicaSetName string
+	Namespace      string
+	PodName        string
+	Container      string
+}
+
+type Resource struct {
+	CPU    float64 `json:"cpu" yaml:"cpu"`
+	Memory float64 `json:"memory" yaml:"memory"`
+}
+
+func (k *Kubernetes) CreatePod(service ServiceTier) {
+	k.ReplicaSetName = service.ServiceName + "-" + generateK8sName(10)
+	k.PodName = k.ReplicaSetName + "-" + generateK8sName(5)
+	k.Namespace = service.ServiceName
+	k.Container = service.ServiceName
+}
+
+func (k *Kubernetes) GetK8sTags() map[string]string {
+	// ref: https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/resource/semantic_conventions/k8s.md
+	return map[string]string{
+		"k8s.cluster.name":   k.ClusterName,
+		"k8s.pod.name":       k.PodName,
+		"k8s.namespace.name": k.Namespace,
+		"k8s.container.name": k.Container,
+	}
+}
+
+func (k *Kubernetes) GenerateMetrics(service ServiceTier) []Metric {
+	if k.ClusterName == "" {
+		return nil
+	}
+
+	var metrics []Metric
+
+	metrics = append(metrics, Metric{
```

Comment:
nit: can we inline all these appends so that it's just `metrics := []Metric{...}`?

Reply:
That's a really good idea, thanks for point it out :)